### PR TITLE
Add service auth header validation to ocr validation endpoint

### PR DESCRIPTION
**Pull request by aliveni**
duplicate of #25

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -12,3 +12,10 @@ management:
 spring:
   application:
     name: Bulk Scan CCD Event Handler Sample Application
+
+idam:
+  s2s-auth:
+    url: ${S2S_URL:http://localhost:4552}
+    secret: ${S2S_SECRET:AAAAAAAAAAAAAAAA}
+
+allowed-services: bulk_scan_processor,bulk_scan_processor_test  # comma separated list of service names
```
Comment (rkondratowicz) [inline src/main/resources/application.yaml]: could be a yaml list

Comment (aliveni) [reply src/main/resources/application.yaml]: just pushed the change. (I thought I already did. I'm creating another pr for controller integration-test)

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/AuthService.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.services.exception.UnauthenticatedException;
+
+import static java.util.Arrays.asList;
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+@Component
+public class AuthService {
+
+    private final AuthTokenValidator authTokenValidator;
+    private final String[] allowedServices;
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/AuthService.java]: would a `List` work?

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/AuthService.java]: I had to change the YAML property value to comma-separated values. The YAML list property is not working with `@Value` annotation. 
A property having `comma-separated values` can be mapped with `String[]` so no need to use utility methods to convert from `String` to `Collection/Array`.

**Comment by aliveni** [comment]
Closing this pr due to the build pipeline issue.

**Pull request closed without merging**